docs/src/main/sphinx/admin/fault-tolerant-execution.rst

```diff
@@ -0,0 +1,338 @@
+========================
+Fault-tolerant execution
+========================
+
+By default, if a Trino node lacks the resources to execute on a task or
+otherwise fails during query execution, the query fails and must be run again
+manually. The longer the runtime of a query, the more likely it is to be
+susceptible to such failures.
+
+Fault-tolerant execution is a mechanism in Trino that enables a cluster to
+mitigate query failures by retrying queries or their component tasks in
+the event of failure. With fault-tolerant execution enabled, in-progress query
+data is spooled and can be re-used by another worker in the event of a worker
+outage or other fault during query execution.
+
+.. note::
+
+    Fault tolerance does not apply to user error. For example, a query that
+    fails because its SQL cannot be parsed is not fault-tolerant.
```

Comment:
It feel weird for me to say that query is is not fault tolerant. The way it is executed is or is not.
And in this case we want to say that if query is inherently broken (cannot be executed because it is misdefined) we will not loop trying to retry it (or retry its tasks).

Reply:
Makes sense, reworded